File: packages/python/src/alumnium/tools/drag_slider_tool.py
from pydantic import Field

from alumnium.drivers.base_driver import BaseDriver

from .base_tool import BaseTool


class DragSliderTool(BaseTool):
    """Set slider to a desired value by clicking at the calculated position based on the slider's width and step."""

    id: int = Field(description="Identifier (ID) of the slider element")
    value: float = Field(description="Desired value to set the slider to")

    def invoke(self, driver: BaseDriver):
        driver.drag_slider(self.id, self.value)

File: packages/typescript/src/tools/DragSliderTool.ts
import { BaseDriver } from "../drivers/BaseDriver.ts";
import { BaseTool } from "./BaseTool.ts";
import { field, type FieldMetadata } from "./Field.ts";

export class DragSliderTool extends BaseTool {
  static description =
    "Set slider to a desired value by clicking at the calculated position based on the slider's width and step.";
  static fields: FieldMetadata[] = [
    field({
      name: "id",
      type: "integer",
      description: "Identifier (ID) of the slider element",
    }),
    field({
      name: "value",
      type: "number",
      description: "Desired value to set the slider to",
    }),
  ];

  id: number;
  value: number;

  constructor(args: { id: number; value: number }) {
    super();
    this.id = args.id;
    this.value = args.value;
  }

  async invoke(driver: BaseDriver): Promise<void> {
    await driver.dragSlider(this.id, this.value);
  }
}
